cfme/v2v/infrastructure_mapping.py

```diff
@@ -0,0 +1,304 @@
+import attr
+from navmazing import NavigateToAttribute
+from navmazing import NavigateToSibling
+from widgetastic.utils import ParametrizedLocator
+from widgetastic.utils import WaitFillViewStrategy
+from widgetastic.widget import ParametrizedView
+from widgetastic.widget import View
+from widgetastic_patternfly import Button
+from widgetastic_patternfly import SelectorDropdown
+from widgetastic_patternfly import Text
+from widgetastic_patternfly import TextInput
+
+from cfme.base.login import BaseLoggedInPage
+from cfme.exceptions import ItemNotFound
+from cfme.modeling.base import BaseCollection
+from cfme.modeling.base import BaseEntity
+from cfme.utils.appliance.implementations.ui import CFMENavigateStep
+from cfme.utils.appliance.implementations.ui import navigate_to
+from cfme.utils.appliance.implementations.ui import navigator
+from cfme.utils.update import Updateable
+from cfme.utils.version import Version
+from cfme.utils.version import VersionPicker
+from widgetastic_manageiq import InfraMappingList
+from widgetastic_manageiq import InfraMappingTreeView
+from widgetastic_manageiq import MultiSelectList
+from widgetastic_manageiq import V2VPaginatorPane
+
+
+class InfrastructureMappingView(BaseLoggedInPage):
+    title = Text("#explorer_title_text")
+
+    @property
+    def in_explorer(self):
+        return self.logged_in_as_current_user and (
+            self.navigation.currently_selected
+            == ["Compute", "Migration", "Infrastructure Mappings"]
+        )
+
+    @property
+    def is_displayed(self):
+        return self.in_explorer
+
+
+class InfraMappingCommonButtons(InfrastructureMappingView):
+    back_btn = Button("Back")
+    next_btn = Button("Next")
+    cancel_btn = Button("Cancel")
+    save = Button("Save")
+    create = Button("Create")
+    add_mapping = Button("Add Mapping")
+    remove_mapping = Button("Remove Selected")
+    remove_all_mappings = Button("Remove All")
+    mappings_tree = InfraMappingTreeView(tree_class="treeview")
+
+
+class ComponentView(View):
+    buttons = View.include(InfraMappingCommonButtons, use_parent=True)
+
+    @ParametrizedView.nested
+    class MappingFillView(ParametrizedView):
+        PARAMETERS = ("object_type",)
+
+        source = MultiSelectList(ParametrizedLocator("source_{object_type}"))
+        target = MultiSelectList(ParametrizedLocator("target_{object_type}"))
+        fill_strategy = WaitFillViewStrategy("15s")
+
+        def after_fill(self, was_change):
+            if not self.parent.add_mapping.disabled:
+                self.parent.add_mapping.click()
+
+    def fill(self, values):
+        was_change = True
+        for mapping in values:
+            self.MappingFillView(self.comp_name).fill(mapping)
```

Comment:
For completeness, please define `ComponentView.comp_name` as 'None' above, so the class has the attribute set. The nested definitions include it, but the base class does not.